wicker/core/datasets.py

```diff
@@ -151,55 +249,55 @@ def get_folder_size(bucket, prefix):
 
         return get_folder_size(bucket, key)
 
-    def _get_dataset_size(self):
-        """Gets total size of the dataset in bits
+    def _get_dataset_size(self, entire_dataset_parquet_dir_size: bool = False):
+        """Gets total size of the dataset in bits.
+
+        Args:
+            entire_dataset_parquet_dir_size (bool): Whether to get the entire parquet dir or just
+                the given partition dir size.
 
         Returns:
             int: total dataset size in bits
         """
+
         # intialize with size of parquet dir
-        par_dir_bytes = self._get_parquet_dir_size()
+        print("Parsing parquet and arrow dir for size.")
+        par_dir_bytes = self._get_parquet_dir_size(entire_dataset_parquet_dir_size=entire_dataset_parquet_dir_size)
 
         # need to know which columns are heavy pntr columns we go to for
-        # byte adding
+        # byte adding, parse which are heavy pointers based off metadata
+        # baked into schema
         schema = self.schema()
         heavy_pointer_cols = []
         for col_name in schema.get_all_column_names():
-            if schema.get_column(col_name).is_heavy_pointer:
+            column = schema.get_column(col_name)
+            if column is not None and column.is_heavy_pointer:
                 heavy_pointer_cols.append(col_name)
 
-        # Each individual row only knows which column file it goes to, so we have to
-        # neccesarily parse all rows :( to get the column files. This should be cached
-        # as metadata but that would require re-curating the datasets.
+        # create arrow table for parsing
         print("Creating arrow table")
         arrow_table = self.arrow_table()
 
         buckets_keys = set()
-        worker = ShuffleWorker(storage=self._storage)
```

Comment:
FYI for Alborz that this `ShuffleWorker` reference above was not actually being used in this routine except go get the `s3_path_factory` reference that is already available on `self`, so removing the `ShuffleWorker` reference is a nice simplification.